docs/ros-snap.md

```diff
@@ -0,0 +1,355 @@
+    [INFO] [WallTime: 1449512660.516307] I heard Hello world 4
+    [ INFO] [1449512660.615954473]: Hello world 5
+    [INFO] [WallTime: 1449512660.616306] I heard Hello world 5
+    ...
+
+### Put it all in a .snap
+
+Create a `snapcraft.yaml` in the workspace root, containing the following:
+
+```yaml
+name: ros-talker-and-listener
+version: 1.0
+summary: ROS Example
+description: Contains talker/listener ROS packages and a .launch file.
+
+binaries:
+  launch_project:
+    exec: roslaunch listener talk_and_listen.launch
+
+parts:
+  foo:
+    plugin: catkin
+    source: .
+    catkin-packages:
+      - talker
+      - listener
+    stage-packages:
+      - ros-indigo-ros-core
+```
+
+Most of this file should look familiar to you, but let's focus on a few specific
+pieces.
+
+```yaml
+# ...
+binaries:
+  launch_project:
+    exec: roslaunch listener talk_and_listen.launch
+# ...
+```
+
+Even though the `talker` and `listener` packages will be installed, they'll be
+within a confined ROS installation, so the user won't be able to simply call
+`rosrun` or `roslaunch`. Instead, you have control over how your .snap is used,
+and here we specify that we only want a single binary, called "launch_project",
+which results in the `roslaunch` call you see. If this seems confusing now, it
+will make more sense when we actually use it.
+
+```yaml
+# ...
+parts:
+  foo:
+    plugin: catkin
+    source: .
+    catkin-packages:
+      - talker
+      - listener
+    stage-packages:
+      - ros-indigo-ros-core
+# ...
+```
+
+This is specifying that the .snap is made up of a single part, which utilizes
+the Catkin plugin. It states that the workspace is in the same path as the
+`snapcraft.yaml`, and it specifies which ROS packages should be included in the
+.snap (`talker` and `listener`). Finally, and this is important, it specifies
+that the Ubuntu package containing `roscore` (ros-indigo-ros-core) should be
+installed into the .snap.
+
+That last point is worth discussing. Currently, since .snaps cannot depend upon
+each other, any .snap that uses roscore must distribute roscore within it. We're
+working on some new features that will enable sharing roscore between snaps, but
+until then there are some limitations to keep in mind:
+
+- `roscore` must be bundled into each .snap that requires it.
+- Port negotiation for `roscore` has not been completed, which means that only a
+  single .snap that runs `roscore` can be installed at a time or they will
+  fight for the same port.
+
+Now that we understand the `snapcraft.yaml`, let's create the .snap!
+
+    $ snapcraft
+
+This will take some time to pull down the dependencies etc., but in the end
+you'll have a .snap.
+
+
+### Take the .snap for a test drive
+
+Transfer the newly-created .snap to your Ubuntu Core machine or instance, and
+install it. For example:
+
+    $ sudo snappy install ros-talker-and-listener_1.0_amd64.snap \
+      --allow-unauthenticated
+
```

Comment:
Probably. I've updated the document to use `snappy-remote`. Does that seem simpler?